libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -440,47 +440,29 @@ async def agenerate(
             ],
             return_exceptions=True,
         )
-        exceptions = []
-        for i, res in enumerate(results):
-            if isinstance(res, BaseException):
-                if run_managers:
-                    await run_managers[i].on_llm_error(
-                        res, response=LLMResult(generations=[])
-                    )
-                exceptions.append(res)
-        if exceptions:
-            if run_managers:
-                await asyncio.gather(
-                    *[
-                        run_manager.on_llm_end(
-                            LLMResult(
-                                generations=[
-                                    cast(ChatResult, res).generations,
-                                ],
-                                llm_output=cast(ChatResult, res).llm_output,
-                            )
-                        )
-                        for run_manager, res in zip(run_managers, results)
-                        if not isinstance(res, Exception)
-                    ]
-                )
-            raise exceptions[0]
-        flattened_outputs = [
-            LLMResult(generations=[res.generations], llm_output=res.llm_output)
-            for res in cast(List[ChatResult], results)
-        ]
-        llm_output = self._combine_llm_outputs(
-            [res.llm_output for res in cast(List[ChatResult], results)]
-        )
-        generations = [res.generations for res in cast(List[ChatResult], results)]
-        output = LLMResult(generations=generations, llm_output=llm_output)
-        await asyncio.gather(
-            *[
-                run_manager.on_llm_end(flattened_output)
-                for run_manager, flattened_output in zip(
-                    run_managers, flattened_outputs
+        # report results and errors
+        if run_managers:
+            jobs = [
```

Comment:
confirm with @nfcampos that reporting results and exceptions together like this isn't a breaking change